### PR TITLE
[WIP] Update AppArmor profile

### PR DESCRIPTION
**Pull request by mudler**
As this is happening on latest TW upgrades, opened a PR with the work of @mnowaksuse, thanks for looking into it.

This work is from @mnowaksuse, so credits goes to him :smiley: 

See: https://progress.opensuse.org/issues/17620

### CODE REVIEW

File: profiles/apparmor.d/usr.share.openqa.script.worker [hunk 1]
```diff
@@ -119,6 +119,8 @@
 
 
   profile /usr/bin/Xvnc {
+    /home/newman/openqa/.../os-autoinst/bmwqemu.pm r,
```
Comment (mnowaksuse) [inline profiles/apparmor.d/usr.share.openqa.script.worker]: This looks odd...

Comment (mudler) [reply profiles/apparmor.d/usr.share.openqa.script.worker]: indeed, possibly also https://github.com/os-autoinst/openQA/pull/1599/files#diff-d9bf49a0805be292590c49c2cc771821R214

File: profiles/apparmor.d/usr.share.openqa.script.worker [hunk 2]
```diff
@@ -165,3 +167,51 @@
...
+    #include <abstractions/openssl>
+
+    /bin/bash rix,
+    /dev/ptmx rw,
+    /etc/ssh/ssh_config r,
+    /etc/x3270/ibm_hosts r,
+    /proc/*/fd/ r,
+    /proc/filesystems r,
+    /proc/meminfo r,
+    /usr/bin/ssh rix,
```
Comment (coolo) [inline profiles/apparmor.d/usr.share.openqa.script.worker]: the problem is if you allow the openQA worker to ssh you can just as well leave off apparmor

File: profiles/apparmor.d/usr.share.openqa.script.worker [hunk 3]
```diff
@@ -165,3 +167,51 @@
 
   }
 }
+
+  profile /usr/bin/Xvnc {
```
Comment (coolo) [inline profiles/apparmor.d/usr.share.openqa.script.worker]: these profiles are duplicated - can we close this PR?

**Comment by okurz** [comment]
So what is the correct approach, create an svirt specific systemd service and let that conflict with apparmor and explain the user that they don't work together?

**Comment by coolo** [comment]
the apparmor service is not the problem - you can leave apparmor enabled for the rest of your system. But if you want to use svirt (or ipmi or mostly !qemu backends) you need to set the worker profile to complain, i.e.
aa-complain /usr/share/openqa/script/worker

**Comment by okurz** [comment]
I see, and can this be incorporated automatically somewhere, maybe even within the specific backends? It's not the same repo where the apparmor profiles reside though

**Comment by coolo** [comment]
You're asking if applications can disable their apparmor profiles? Think about it!

**Comment by okurz** [comment]
:D no, not that. I mean abort with a helpful error and leave the disabling to the user. Something like "svirt backends act like Trojan horsis, they thrive for world domination through ssh. If you want to be Helena, disable apparmor" or something like that

File: profiles/apparmor.d/usr.share.openqa.script.worker (hunk 3 above)
Comment (mudler) [reply profiles/apparmor.d/usr.share.openqa.script.worker]: Sure, it was here just to have a better insight of the proposed changes in the progress issue

**Comment by mudler** [comment]
This can be closed now - @okurz feel free to open a PR if you have a solution that does not involve dynamic rewrital of apparmor rules from the same application

**Comment by SergioAtSUSE** [comment]
Related: https://github.com/os-autoinst/openQA/pull/1645

**Pull request closed without merging**